apollo-federation/src/query_graph/extract_subgraphs_from_supergraph.rs

```diff
@@ -302,6 +309,22 @@ struct TypeInfos {
     input_object_types: Vec<TypeInfo>,
 }
 
+fn get_original_directive_names(
```

Comment:
Do the changes in this file do everything we need for expansion (i.e., we're not going to break anything)? Also, do we need to update our docs to state that we don't support these new ones?

Reply:
https://preview-docs.apollographql.com/graphos/connectors/limitations#interactions-with-graphos-router-features already mentions demand control, do you think we should be more specific about the directives?

Reply:
these changes look orthogonal, since cost/listSize are encoded specifically into the supergraph, as opposed to our general join__directive approach